src/GC/GC.jl

```diff
@@ -3,77 +3,134 @@
 
 Garbage collection of Python objects.
 
-See `disable` and `enable`.
+See [`gc`](@ref).
 """
 module GC
 
 using ..C: C
 
-const ENABLED = Ref(true)
-const QUEUE = C.PyPtr[]
+const QUEUE = (; items = C.PyPtr[], lock = Threads.SpinLock())
+const HOOK = WeakRef()
 
 """
     PythonCall.GC.disable()
 
-Disable the PythonCall garbage collector.
+Do nothing.
 
-This means that whenever a Python object owned by Julia is finalized, it is not immediately
-freed but is instead added to a queue of objects to free later when `enable()` is called.
+!!! note
 
-Like most PythonCall functions, you must only call this from the main thread.
+    Historically this would disable the PythonCall garbage collector. This was required
+    for safety in multi-threaded code but is no longer needed, so this is now a no-op.
 """
-function disable()
-    ENABLED[] = false
-    return
-end
+disable() = nothing
```

Comment:
```suggestion
@deprecate disable() identity(nothing) false
```

This would be the documented equivalent. But normally if there's no alternative function to call you just write a message manually with `Base.depwarn`: https://github.com/search?q=/Base.depwarn/%20language:julia&type=code

Reply:
(It's helpful for library maintainers, since the `depwarn` will show up during unittests but not during normal execution – so I know what functions I need to update.)

Reply:
Since if you do it with the `@deprecated` API, it doesn't really make sense in this context:

```julia
julia> function foo()
       end
foo (generic function with 1 method)

julia> function _foo()
       end
_foo (generic function with 1 method)

julia> @deprecate foo() _foo()
foo (generic function with 1 method)

julia> foo()
┌ Warning: `foo()` is deprecated, use `_foo()` instead.
│   caller = top-level scope at REPL[4]:1
└ @ Core REPL[4]:1
```